src/reactviews/pages/PublishProject/publishProjectStateProvider.tsx

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { createContext, useMemo } from "react";
+import { useVscodeWebview2 } from "../../common/vscodeWebviewProvider2";
+import { getCoreRPCs2 } from "../../common/utils";
+import { WebviewRpc } from "../../common/rpc";
+import {
+    PublishDialogReducers,
+    PublishDialogState,
+    IPublishForm,
+    PublishProjectProvider,
+} from "../../../sharedInterfaces/publishDialog";
+import { FormEvent } from "../../../sharedInterfaces/form";
+import {
+    LoggerLevel,
+    WebviewTelemetryActionEvent,
+    WebviewTelemetryErrorEvent,
+} from "../../../sharedInterfaces/webview";
+
+export interface PublishProjectContextProps extends PublishProjectProvider {
+    readonly state?: PublishDialogState;
+    log(message: string, level?: LoggerLevel): void;
+    sendActionEvent(event: WebviewTelemetryActionEvent): void;
+    sendErrorEvent(event: WebviewTelemetryErrorEvent): void;
+    /** Advanced escape hatch; prefer using typed provider methods */
+    extensionRpc: WebviewRpc<PublishDialogReducers>;
+}
+
+// Optional payload for publishNow future expansion
+export type PublishNowPayload = Parameters<PublishProjectProvider["publishNow"]>[0];
+
+export const PublishProjectContext = createContext<PublishProjectContextProps | undefined>(
+    undefined,
+);
+
+export const PublishProjectStateProvider: React.FC<{ children: React.ReactNode }> = ({
+    children,
+}) => {
+    const { extensionRpc, getSnapshot } = useVscodeWebview2<
+        PublishDialogState,
+        PublishDialogReducers
+    >();
+
+    const value = useMemo<PublishProjectContextProps>(
+        () => ({
+            get state() {

```

Comment:
Please remove state from here and just use selector. Having a centralized state defeats the purpose of having selector.